pytups/superdict.py

```diff
@@ -454,16 +460,21 @@ def kapply(self, func: Callable, *args, **kwargs) -> "SuperDict":
         """
         return SuperDict({k: func(k, *args, **kwargs) for k in self})
 
-    def sapply(self, func: Callable, other: dict, *args, **kwargs) -> "SuperDict":
+    def sapply(
+        self, func: Callable, other: Union[dict, int, float], *args, **kwargs
+    ) -> "SuperDict":
         """
         Applies function to both dictionaries.
         Using keys of the self.
         It's like applying a function over the left join.
 
         :param callable func: function to apply.
-        :param dict other: dictionary to apply function to
+        :param Union[dict, int, float] other: either ar int or float to perform the operation over or another dictionary
         :return: new :py:class:`SuperDict`
         """
+        if isinstance(other, (int, float)):
```

Comment:
I prefer the code in your comment:

```python

if isinstance(other, (int, float):
  return SuperDict({k: func(v, other, *args, **kwargs) for k, v in self.items()})

```

Better yet, could we just better do it with vapply for simplicity, as you mentioned in the example?

```
d1 = d1.vapply(lambda v: v+3)
```

Maybe we should also do this for the regular operation. Just use the `kvapply` instead of the dict comprehension:

```python
return v.kvapply(lambda k, v: func(v, other[k], *args, **kwargs))

```